src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/TypePublisherSource.cs

```diff
@@ -17,6 +17,14 @@ public TypePublisherSource(Type messageType, PublisherAddress address)
 
         public IEnumerable<PublisherTableEntry> Generate(Conventions conventions)
         {
+            if (!conventions.IsMessageType(messageType))
+            {
+                throw new Exception($"Cannot configure publisher for type {messageType.FullName} because it is not considered a message. Message types have to either implement NServiceBus.IMessage interface or follow a defined message convention.");
+            }
+            if (!conventions.IsEventType(messageType))
```

Comment:
As I eluded to above I wonder if this should be more relaxed and be a `conventions.IsCommandType()` instead to not block users that only use the message convention?